src/App.vue

```diff
@@ -358,48 +359,22 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
         throw error // go to catch()
       })
 
-      // fetch NR data
-      const nrResponse = await this.fetchNameRequest(this.nrNumber).catch(error => {
-        console.log('NR error =', error) // eslint-disable-line no-console
-        this.nameRequestInvalidErrorDialog = true
-        throw error // go to catch()
-      })
-
-      // ensure NR was found
-      if (!nrResponse) {
-        this.nameRequestInvalidType = NameRequestStates.NOT_FOUND
-        this.nameRequestInvalidErrorDialog = true
-        return // go to finally()
-      }
-
-      // ensure NR is valid
-      if (!this.isNrValid(nrResponse)) {
-        this.nameRequestInvalidType = NameRequestStates.INVALID
-        this.nameRequestInvalidErrorDialog = true
-        return // go to finally()
-      }
-
-      // ensure NR is consumable
-      const state = this.getNrState(nrResponse)
-      if (!state || state !== NameRequestStates.APPROVED) {
-        this.nameRequestInvalidType = state || NameRequestStates.INVALID
-        this.nameRequestInvalidErrorDialog = true
-        return // go to finally()
-      }
-
-      // if we get this far, the NR is good to go!
-      nrResponse.isConsumable = true
-
-      this.setNameRequestState(this.generateNameRequestState(nrResponse, null))
-
       try {
+        this.setTemporaryId(this.tempId)
         // Retrieve draft filing if it exists for the NR Number specified
-        const draftFiling = await this.fetchDraft()
+        let draftFiling = await this.fetchDraft()
+        let emptyFiling = this.buildEmptyFiling()
+        if (draftFiling?.incorporationApplication?.nameRequest) {
+          await this.populateNameRequest(draftFiling)
+          emptyFiling = this.buildFiling()
+        }
+
+        draftFiling = { ...emptyFiling.filing, ...draftFiling }
```

Comment:
👍

Reply:
Although, it would be nice if emptyFiling was filing (instead of containing it). Possible?

Reply:
because buildFiling() is called elsewhere I'm a little hesitant to do this

Reply:
OK, nvm